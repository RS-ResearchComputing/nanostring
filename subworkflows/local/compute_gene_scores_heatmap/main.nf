//
// Compute gene scores and plot heatmap
//

//
// MODULES
//
include { COMPUTE_GENE_SCORES     } from '../../../modules/local/compute_gene_scores'
include { CREATE_GENE_HEATMAP     } from '../../../modules/local/create_gene_heatmap'

workflow COMPUTE_GENE_SCORES_HEATMAP {
    take:
    normalized_counts      // channel: path(rcc files)
    ch_gene_score_config   // file: /path/to/samplesheet.csv
    annotated_endo_data
    ch_heatmap_genes_to_filter


    main:
    ch_versions = Channel.empty()
    ch_multiqc_files = Channel.empty()

    //
    // MODULE: Compute gene scores for supplied YAML gene score file
    //
    COMPUTE_GENE_SCORES(
        normalized_counts,
        ch_gene_score_config
    )
    ch_versions      = ch_versions.mix(COMPUTE_GENE_SCORES.out.versions)
    ch_multiqc_files = ch_multiqc_files.mix(COMPUTE_GENE_SCORES.out.scores_for_mqc.collect())

    //
    // MODULE: Compute gene-count heatmap for MultiQC report based on annotated (ENDO) counts
    //
    if(!params.skip_heatmap){
        CREATE_GENE_HEATMAP (
            annotated_endo_data,
            normalized_counts,
            ch_heatmap_genes_to_filter.toList()
        )
        ch_versions       = ch_versions.mix(CREATE_GENE_HEATMAP.out.versions)
        ch_multiqc_files  = ch_multiqc_files.mix(CREATE_GENE_HEATMAP.out.gene_heatmap.collect())
    }

    emit:
    versions                = ch_versions                 // channel: [ versions.yml ]
    multiqc_files           = ch_multiqc_files
}
