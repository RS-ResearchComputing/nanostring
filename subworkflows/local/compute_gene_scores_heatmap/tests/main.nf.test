nextflow_workflow {

    name "Test Subworkflow COMPUTE_GENE_SCORES_HEATMAP"
    script "../main.nf"
    workflow "COMPUTE_GENE_SCORES_HEATMAP"

    test("test profile") {
        when {
            workflow {
                """
                input[0] = file(params.pipelines_testdata_base_path + 'nanostring/modules_test_data/test_profile/normalized_counts.tsv', checkIfExists: true )
                input[1] = file(params.pipelines_testdata_base_path + 'nanostring/gene_score_test.yaml', checkIfExists: true )
                input[2] = file(params.pipelines_testdata_base_path + 'nanostring/modules_test_data/test_profile/counts_Norm_GEX_ENDO.tsv', checkIfExists: true )
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
