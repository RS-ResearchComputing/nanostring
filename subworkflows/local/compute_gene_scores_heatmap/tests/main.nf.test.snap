{
    "test profile": {
        "content": [
            {
                "0": [
                    "versions.yml:md5,63b539ddb8d646a7cf480f3de2e47100",
                    "versions.yml:md5,6fac9e04824b7752f45abd6b1d4ef0a9"
                ],
                "1": [
                    [
                        "gene_heatmap_mqc.png:md5,3472b2c3d92a326f82ea7b542163a6c3"
                    ],
                    [
                        "signature_scores_mqc.txt:md5,315dc730c79b02067b542074eaeb07b9",
                        "signature_scores_qc_mqc.txt:md5,88ec48e1d7488357663557b378615923"
                    ]
                ],
                "multiqc_files": [
                    [
                        "gene_heatmap_mqc.png:md5,3472b2c3d92a326f82ea7b542163a6c3"
                    ],
                    [
                        "signature_scores_mqc.txt:md5,315dc730c79b02067b542074eaeb07b9",
                        "signature_scores_qc_mqc.txt:md5,88ec48e1d7488357663557b378615923"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,63b539ddb8d646a7cf480f3de2e47100",
                    "versions.yml:md5,6fac9e04824b7752f45abd6b1d4ef0a9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T21:32:37.651136335"
    }
}